src/derive.rs

```diff
@@ -283,7 +283,7 @@ pub trait Subcommand: FromArgMatches + Sized {
 /// ```
 pub trait ArgEnum: Sized {
     /// All possible argument choices, in display order.
-    const VARIANTS: &'static [&'static str];
+    const VARIANTS: &'static [ArgValue<'static>];
```

Comment:
I wonder if we are getting to the point of re-working this
- It isn't "variants" because we respect "skip", Even more so with this change
- A part of me wonders if we should instead expose this through `IntoIterator`

Reply:
you mean implement IntoIterator for ArgEnums?

Reply:
this would also help with the from reference implementations in #2758.

Reply:
Yes.  To be clear, this is me brainstorming / thinking out loud.  We can probably make minor changes (like the name) for moving this PR forward and defer this to another Issue/PR so this doesn't get slowed down by decision making.

Reply:
I agree with re-thinking this.